src/Ocelot/DependencyInjection/ConfigurationBuilderExtensions.cs

```diff
@@ -28,37 +29,42 @@ public static IConfigurationBuilder AddOcelotBaseUrl(this IConfigurationBuilder
             return builder;
         }
 
-        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder)
+        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, IHostingEnvironment env = null)
         {
-            return builder.AddOcelot(".");
+            return builder.AddOcelot(".", env);
         }
 
-        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, string folder)
+        public static IConfigurationBuilder AddOcelot(this IConfigurationBuilder builder, string folder, IHostingEnvironment env = null)
         {
-            const string pattern = "(?i)ocelot\\.([a-zA-Z0-9]*)(\\.json)$";
+            const string primaryConfigFile = "ocelot.json";
 
-            var reg = new Regex(pattern);
+            const string globalConfigFile = "ocelot.global.json";
 
-            var files = Directory.GetFiles(folder)
-                .Where(path => reg.IsMatch(path))
+            const string subConfigPattern = @"^ocelot\.[a-zA-Z0-9]+\.json$";
+
+            string excludeConfigName = env?.EnvironmentName != null ? $"ocelot.{env.EnvironmentName}.json" : string.Empty;
+
+            var reg = new Regex(subConfigPattern, RegexOptions.IgnoreCase | RegexOptions.Singleline);
+
+            var files = new DirectoryInfo(folder)
+                .EnumerateFiles()
+                .Where(fi => reg.IsMatch(fi.Name) && (fi.Name != excludeConfigName))
                 .ToList();
 
             var fileConfiguration = new FileConfiguration();
 
             foreach (var file in files)
             {
-                // windows and unix sigh...
-                if(files.Count > 1 && (Path.GetFileName(file) == "ocelot.json"))
+                if(files.Count > 1 && file.Name.Equals(primaryConfigFile, StringComparison.OrdinalIgnoreCase))
                 {
                     continue;
                 }
 
-                var lines = File.ReadAllText(file);
+                var lines = File.ReadAllText(file.FullName);
```

Comment:
What does FullName do? :o

Reply:
Well I thought that using more strict FileInfo type instead of storing file path can simplify the code and thus unix/windows specific path handling can be omitted in this case.
https://msdn.microsoft.com/ru-ru/library/system.io.fileinfo_properties(v=vs.110).aspx

But if you want me to minimize the code impact I can just return it to initial state and deal with strings.

Reply:
no thats fine, I just didnt know what it was :)